Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,333 @@
+
+            for (int ix = 0; ix < neutralPoseVertices.Length; ix++)
+            {
+                Vector3 p = neutralPoseVertices[ix];
+
+                handMeshUVs[ix] = new Vector2(p.x * scale + 0.5f, (p.y - minY) * scale);
+            }
+        }
+
+        private async void SetHandMeshObserver(SpatialInteractionSourceState sourceState)
+        {
+            handMeshObserver = await sourceState.Source.TryCreateHandMeshObserverAsync();
+        }
+#endif // WINDOWS_UWP
+
+        #region Update data functions
+
+        /// <inheritdoc />
+        public override void UpdateController(InputDevice inputDevice)
+        {
+            if (!Enabled) { return; }
+
+            base.UpdateController(inputDevice);
+
+            UpdateHandData(inputDevice);
+
+            for (int i = 0; i < Interactions?.Length; i++)
+            {
+                switch (Interactions[i].InputType)
+                {
+                    case DeviceInputType.IndexFinger:
+                        UpdateIndexFingerData(Interactions[i]);
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update the hand data from the device.
+        /// </summary>
+        /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform.</param>
+        private void UpdateHandData(InputDevice inputDevice)
+        {
+            Hand hand;
+            if (inputDevice.TryGetFeatureValue(CommonUsages.handData, out hand))
+            {
+                foreach (HandFinger finger in handFingers)
+                {
+                    if (hand.TryGetFingerBones(finger, fingerBones))
+                    {
+                        for (int i = 0; i < fingerBones.Count; i++)
+                        {
+                            TrackedHandJoint trackedHandJoint = ConvertToTrackedHandJoint(finger, i);
+                            Bone bone = fingerBones[i];
+
+                            Vector3 position = Vector3.zero;
+                            Quaternion rotation = Quaternion.identity;
+
+                            if (bone.TryGetPosition(out position) || bone.TryGetRotation(out rotation))
+                            {
+                                // We want input sources to follow the Playspace, so fold in the playspace transform here to
+                                // put the controller pose into world space.
+                                position = MixedRealityPlayspace.TransformPoint(position);
+                                rotation = MixedRealityPlayspace.Rotation * rotation;
+
+                                unityJointPoses[trackedHandJoint] = new MixedRealityPose(position, rotation);
+                            }
+                        }
+
+                        // Unity doesn't provide a palm joint, so we synthesize one here
+                        unityJointPoses[TrackedHandJoint.Palm] = CurrentControllerPose;
```

Comment:
This needs to be rotated slightly to properly match other joint orientations. Fix incoming

Reply:
Gonna leave this for now. Depends on some XR SDK fixes that haven't shipped yet